generator/readme_generator.py

```
@@ -109,12 +109,10 @@ def generate_readme_interactively(project_path: Path, use_ai: bool) -> str:
     return readme_content
 
 
-def generate_readme_with_llm(user_input: Dict, context: Dict) -> str:
-    """Generate README using Gemini."""
+def generate_readme_with_llm(user_input: Dict, context: Dict, provider: str = 'gemini', api_key: str = None) -> str:
```

Ruff (0.14.14)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)